src/squashmerge.c

```diff
@@ -228,28 +251,28 @@ void* decompress_blocks(void* data)
 
 		if (i % no_threads == id)
 		{
-			size_t offset = ntohl(source_blocks[i].offset);
+			size_t offset = ntohll(source_blocks[i].offset);
```

Comment:
I see mostly `offset` gets assigned to `size_t` variables throughout the code, whose size is platform dependent.
Should we declare offset as `size_t` instead? Or just change all these `size_t` declarations to `uint64_t`? Our robot platforms are 64 bits, but as squashmerge also runs on DUC in Cloud, can we guarantee it will always be a 64 bits system?

If we can guarantee that and that Robot SW will never use a 32 bits system, then we can keep it as `size_t`.

Reply:
Also (minor), few lines below, there's an `fprintf`:
```
					fprintf(stderr, "Block decompression resulted in different size.\n"
									"\toffset: 0x%08lx\n"
									"\tlength: %lu\n"
									"\texpected unpacked length: %lu\n"
									"\treal unpacked length: %lu\n",
							offset, length, unc_length, ret);
```
I'm not sure offset will print correctly being 64 bits with `0x%08lx`.

Reply:
Given that squashdelta and squashmerge use memory maps to read from files, I believe that a >4GB file simply wouldn't work on a 32-bit system, correct?

Reply:
Also on my laptop `size_t` is typedefed to `long unsigned int` and I double-checked the documentation for `%08lx`, and I think it's correct as-is.